dotCMS/src/main/java/com/dotmarketing/cms/urlmap/filters/URLMapFilter.java

```diff
@@ -245,7 +250,12 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
                         if (optSession != null) {
                             optSession.setAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE,
                                     String.valueOf(
-                                            conAPI.find(c.getInode(), user, true).getLanguageId()));
+                                            conAPI.find(c.getInode(), wuserAPI.getSystemUser(), true).getLanguageId()));
+                        }
+                        Contentlet con = conAPI.find(c.getInode(), wuserAPI.getSystemUser(), true);
+                        if(!perAPI.doesUserHavePermission(con, PermissionBitAPIImpl.PERMISSION_READ,user,true)){
+                            response.sendRedirect("/dotCMS/login?referrer="+UtilMethods.encodeURIComponent(url));
+                            return;
```

Comment:
should we have this struts-cms behavior hardcoded? shouldn't we handle this the same way VelocityServlet handles limited assets that require frontend login (also, encoding the referrer)?